### PR TITLE
Add `sparseml.recipe_template` cli

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR adds a CLI for `sparseml.recipe_template` + automated tests

TODO:
- [ ] Change base to `sparsify.recipe_template.dev` before landing

### CODE REVIEW

File: src/sparseml/recipe_template/cli.py [hunk 1]
```diff
@@ -12,15 +14,61 @@
...
+    help="Specify if recipe should include pruning steps, can also take in the "
+    "name of a pruning algorithm",
+)
+@click.option(
+    "--quantization",
+    type=click.Choice(["true", "false"], case_sensitive=False),
+    default="false",
+    help="Specify if recipe should include quantization steps, can also take in "
+    "the name of the target hardware",
+)
```
Comment (corey-nm) [inline src/sparseml/recipe_template/cli.py:43]: Use `is_flag=True, default=False, type=bool`

File: src/sparseml/recipe_template/cli.py [hunk 2]
```diff
@@ -12,15 +14,61 @@
...
+_LOGGER = logging.getLogger(__name__)
 
 
-def main(*args, **kwargs):
+@click.command(context_settings=(dict(show_default=True)))
+@click.version_option(version=__version__)
+@click.option(
+    "--pruning",
+    # other algos will be added in future
+    type=click.Choice(["true", "false", "gmp"], case_sensitive=False),
```
Comment (corey-nm) [inline src/sparseml/recipe_template/cli.py:32]: what's the difference between `true` and `gmp`? do they use different algorithms? if so, recommend:
```
type=click.Choice(["gmp", <OTHER ALG>]),
default=None
```

File: src/sparseml/recipe_template/cli.py [hunk 3]
```diff
@@ -12,15 +14,61 @@
...
+    "--learning_rate",
+    type=click.Choice(
+        ["constant", "cyclic", "stepped", "exponential", "linear"], case_sensitive=False
+    ),
+    default="constant",
+    help="Specify growth function for learning rate",
+)
+@click.option(
+    "--target",
+    type=click.Choice(["vnni", "tensorrt", "default"], case_sensitive=False),
```
Comment (corey-nm) [inline src/sparseml/recipe_template/cli.py:55]: would it be possible to replace default with whatever default means?

File: src/sparseml/recipe_template/cli.py [hunk 4]
```diff
@@ -12,15 +14,61 @@
...
+    "the name of the target hardware",
+)
+@click.option(
+    "--lr",
+    "--learning_rate",
+    type=click.Choice(
+        ["constant", "cyclic", "stepped", "exponential", "linear"], case_sensitive=False
+    ),
+    default="constant",
+    help="Specify growth function for learning rate",
```
Comment (KSGulin) [inline src/sparseml/recipe_template/cli.py:51]: nit
```suggestion
    help="Specify learning rate schedule function",
```

File: src/sparseml/recipe_template/cli.py [hunk 5]
```diff
@@ -12,15 +14,61 @@
...
+
+_LOGGER = logging.getLogger(__name__)
 
 
-def main(*args, **kwargs):
+@click.command(context_settings=(dict(show_default=True)))
+@click.version_option(version=__version__)
+@click.option(
+    "--pruning",
+    # other algos will be added in future
```
Comment (dbogunowicz) [inline src/sparseml/recipe_template/cli.py:31]:
```suggestion
```

**Review by dbogunowicz — COMMENTED** [review]
nice

File: src/sparseml/recipe_template/cli.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparseml/recipe_template/cli.py:43]: Good catch, thanks!

File: src/sparseml/recipe_template/cli.py [hunk 6]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (rahul-tuli) [reply src/sparseml/recipe_template/cli.py:1]: Yes! this script can be invoked directly without explicitly specifying python, `(venv)$ ./cli.py`

File: src/sparseml/recipe_template/cli.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparseml/recipe_template/cli.py]: This is a requirement in the acceptance doc, Corey, will bring it up and make the change if approved

**Pull request closed without merging**